### ISSUE TITLE
The download process seems to have crashed (Error)

### ISSUE THREAD

**Issue by Daigh** [hama3254/Crunchyroll-Downloader-v3.0#991]
![imagen](https://github.com/hama3254/Crunchyroll-Downloader-v3.0/assets/6308133/d4f82065-7d60-40c6-b4e6-8dfb6193ef5f)

I don't know what would cause this error :(

**Comment by Raychan87** [comment]
I have a similar error. The first video is always broken and the second video does not load.
![grafik](https://github.com/hama3254/Crunchyroll-Downloader-v3.0/assets/18511462/c1185208-2297-4569-9402-75be152ad0f8)

**Comment by hama3254** [comment]
Probably the same as #990 which seems to be a DRM issue.
It seems like Crunchyroll updated the switch api to now use DRM streams.

**Comment by Konvinillo** [comment]
> Probablemente lo mismo que [el número 990](https://github.com/hama3254/Crunchyroll-Downloader-v3.0/issues/990) , que parece ser un problema de DRM. Parece que Crunchyroll actualizó la API del conmutador para usar ahora transmisiones DRM.

Does this mean the termination of the unloader?

**Comment by stratuma** [comment]
@hama3254 Crunchyroll killed the switch endpoint. There is no other non-drm endpoint left. RIP

**Comment by vabarboza** [comment]
I did a download test with the trial version of StreamFab Crunchyroll Downloader, and I was able to download it, I believe they use another way to download the files

**Comment by Elwador** [comment]
@vabarboza yes they download the drm protected files and decrypt them

**Comment by vabarboza** [comment]
@Elwador So is it possible to modify the application to do the same?

**Comment by Elwador** [comment]
@vabarboza possible yes but @hama3254 already said a couple of times that he won't do it 

you can also see this text at the top of the [Readme](https://github.com/hama3254/Crunchyroll-Downloader-v3.0/blob/master/README.md) "Crunchyroll-Downloader-v3.0 is deprecated since Crunchyroll updated to a DRM only system."

that's from the last time everyone thought crunchyroll had no endpoint left that wasn't drm protected

also i wouldn't suggest to spend money on StreamFab as long as there are free versions on github that you can use

**Comment by hama3254** [comment]
> also i wouldn't suggest to spend money on StreamFab as long as there are free versions on github that you can use

I would suggest to use OBS and another browser as the default you are using to record the stream rather than paying for a downloading tool.

More about that here: https://github.com/hama3254/Crunchyroll-Downloader-v3.0/issues/911#issuecomment-2002954344

**Comment by stratuma** [comment]
> I would suggest to use OBS and another browser as the default you are using to record the stream rather than paying for a downloading tool.

You need to disable hardware acceleration in the browser to be able to record.

**Comment by stratuma** [comment]
I would suggest to get L3 widevine keys (its not hard at all) and decrypt the drm videos from crunchyroll.

**Comment by TaMaGo-TTV** [comment]
> I would suggest to get L3 widevine keys (its not hard at all) and decrypt the drm videos from crunchyroll.

is that legall and can i get baned from crunchy?

**Comment by stratuma** [comment]
> Is that legal, and can I get banned from crunchy?

Its not legal but you don't get banned

**Comment by TaMaGo-TTV** [comment]
> > Is that legal, and can I get banned from crunchy?
> 
> Its not legal but you don't get banned

Okay and how can they "catch" me? should I just use my vpn and im safe?

**Comment by stratuma** [comment]
> Okay and how can they "catch" me? should I just use my vpn and im safe?

They are not going to catch you for decrypting a anime episode xD
They can't even track it correctly

**Comment by TaMaGo-TTV** [comment]
> They are not going to catch you for decrypting a anime episode xD They can't even track it correctly

Bro I live in Germany so who knows XD

**Comment by stratuma** [comment]
> Bro I live in Germany so who knows XD

You can decide if you want to use a drm downloader or not, its your decision

**Comment by TaMaGo-TTV** [comment]
> > Bro I live in Germany so who knows XD
> 
> You can decide if you want to use a drm downloader or not, its your decision

I mean there is no choice if I want to download anime from crunchy

**Comment by stratuma** [comment]
> I mean there is no choice if I want to download anime from crunchy

Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0

Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.

**Comment by TaMaGo-TTV** [comment]
> > I mean there is no choice if I want to download anime from crunchy
> 
> Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0
> 
> Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.

Ja, ich hab den schon drauf ich wusste, nur nicht wie Safe es ist mit diesem L3 widevine Zeug.

Aber wenn du sagst, dass die einen schlecht tracken können, dann würde ich das mal probieren.

**Comment by Jaynator495** [comment]
> > > I mean there is no choice if I want to download anime from crunchy
> > 
> > 
> > Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0
> > Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.
> 
> Ja, ich hab den schon drauf ich wusste, nur nicht wie Safe es ist mit diesem L3 widevine Zeug.
> 
> Aber wenn du sagst, dass die einen schlecht tracken können, dann würde ich das mal probieren.

You can also take a look at multi-download-nx, it offers some other services that is in Germany like ADN as well

**Comment by hama3254** [comment]
> > Is that legal, and can I get banned from crunchy?
> 
> Its not legal but you don't get banned

Consequences are unlikely to happen for users, I am more worried about the developers.

I personally already got a cease and desist per Mail from a right holder of some anime (not CR).
Not to important since they called me `Dear Sir` so I ignored the 2nd deadline since the first one was already over.

The only question I would ask myself, if I breake copyrighted law anyway why breake DRM on Crunchyroll that cost money to access most shows if we can sail the high seas and rip from them.
If I understand it correctly, both options are equal illegal.

**Comment by AsIFall** [comment]
> > I mean there is no choice if I want to download anime from crunchy
> 
> Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0
> 
> Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.

Ich habe trotz des Guides knapp sechs Stunden gebraucht um diese Schlüssel generieren zu können.
Der Guide war auf der einen Seite okay, an manchen Stellen wohl nicht aktualisiert.
Wenn ihr zu dem Punkt mit dem "CMD" kommt. Vor dem zweiten Öffnen muss "pycryptodome" und die Version 3.20.0 von protobus installiert werden. Das wird in dem Android Emulator gemacht. Da muss dann mithilfe von pip install protobus==3.20.0 eingegeben und bestätigt werden. Wenn eine neue Version von protobus installiert wird, funktioniert "dump_keys.py" nicht.

Ich kann deine Version des Downloaders aber nur scheinbar verwenden. Er lädt das Video, bricht den Audio-Download aber ab, sodass das Video letztendlich nicht gespeichert werden kann. Habe versucht, es als MP4- bzw. als Matroschka (mkv) herunterzuladen.

**Comment by stratuma** [comment]
Wenn der guide zu schwer ist kann ich dir nur empfehlen einen anderen zu suchen.
Alternativ wäre dieser guide: https://simpcity.su/threads/guide-for-getting-drm-content-wip.181199/

**Comment by TaMaGo-TTV** [comment]
> Wenn der guide zu schwer ist kann ich dir nur empfehlen einen anderen zu suchen. Alternativ wäre dieser guide: https://simpcity.su/threads/guide-for-getting-drm-content-wip.181199/

Ist der Key aber dann nur für eine Folge und ich muss für jede folge einen Key erstellen?

**Comment by stratuma** [comment]
> Ist der Key aber dann nur für eine Folge und ich muss für jede folge einen Key erstellen?

Nein wenn du die keys nicht sharest dann kannst du sie für alles was L3 Encrypted ist hernehmen und das für Jahre ohne neue besorgen zu müssen

**Comment by TaMaGo-TTV** [comment]
> > Ist der Key aber dann nur für eine Folge und ich muss für jede folge einen Key erstellen?
> 
> Nein wenn du die keys nicht sharest dann kannst du sie für alles was L3 Encrypted ist hernehmen und das für Jahre ohne neue besorgen zu müssen

Okay check🤝🏻

Ich hatte gestern Probleme einen key zu erstellen, ich versuchs mal mut dem anderen Tutorial.

**Comment by hama3254** [comment]
An alle bitte haltet die Diskussion auf dem Thema, das umgehend von DRM ist nicht Teil dieses Projektes.

**Comment by AsIFall** [comment]
> An alle bitte haltet die Diskussion auf dem Thema, das umgehend von DRM ist nicht Teil dieses Projektes.

Sorry.

**Comment by stratuma** [comment]
> An alle bitte haltet die Diskussion auf dem Thema, das umgehend von DRM ist nicht Teil dieses Projektes.

Ok

**Comment by stratuma** [comment]
@hama3254 Wir suchen momentan noch accesspoints aber es scheint keine non-drm accesspoints mehr zu geben.

**Comment by TaMaGo-TTV** [comment]
> @hama3254 Wir suchen momentan noch accesspoints aber es scheint keine non-drm accesspoints mehr zu geben.

Warum macht crunchy das eigentlich? Haben die in deren eigenen AGBs nicht etwas von Download to Own stehen? Oder gilt das nur für die In App Offline Downloads?

**Comment by Jaynator495** [comment]
> > Warum macht crunchy das eigentlich? Haben die in deren eigenen AGBs nicht etwas von Download to Own stehen? Oder gilt das nur für die In App Offline Downloads?
> 
> Crunchyroll macht das weil sie geldgeil sind. Aber wie das sprichwort "if buying isn't owning, piracy isn't stealing" schon sagt, machen sie sich damit nur nochmehr Feinde.

Lots of deepl later, and I completely agree. I've very against this whole new DRM paradigm that everyone and their dog thinks they need. I'm not sure why people and companies don't realize by Gabe Newell's success that the answer is to improve user experience, not muddle it with all these restrictions. In the case of crunchy, they are actively turning away the subscribers that want to use their service and are willing to pay for it, but for whatever reason can't (be it slow internet or whatever else). Sure in the short term they might have a slight loss, but in the long term it leads to a larger gain.

**Comment by Jaynator495** [comment]
> > > I mean there is no choice if I want to download anime from crunchy
> > 
> > 
> > Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0
> > Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.
> 
> OR
> 
> https://github.com/Crunchy-DL/Crunchy-Downloader

well as long as we're all actually linking the projects, 
https://github.com/anidl/multi-downloader-nx

**Comment by NickQ29** [comment]
> > > > I mean there is no choice if I want to download anime from crunchy
> > > 
> > > 
> > > Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0
> > > Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.
> > 
> > 
> > OR
> > https://github.com/Crunchy-DL/Crunchy-Downloader
> 
> well as long as we're all actually linking the projects, https://github.com/anidl/multi-downloader-nx

Can anyone explain, in terms appropriate for someone with minus infinity tech-savvy, what the difference is between all of these and which one is the easiest to make work for, again, someone with infinitely zero capability for tech-savvy?

Because the first one has links to "explanation" threads that are complete and utter gibberish of the kind I seek out GUIs specifically to avoid having to ever deal with.

**Comment by petenun** [comment]
> > > > > I mean there is no choice if I want to download anime from crunchy
> > > > 
> > > > 
> > > > Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0
> > > > Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.
> > > 
> > > 
> > > OR
> > > https://github.com/Crunchy-DL/Crunchy-Downloader
> > 
> > 
> > well as long as we're all actually linking the projects, https://github.com/anidl/multi-downloader-nx
> 
> Can anyone explain, in terms appropriate for someone with minus infinity tech-savvy, what the difference is between all of these and which one is the easiest to make work for, again, someone with infinitely zero capability for tech-savvy?
> 
> Because the first one has links to "explanation" threads that are complete and utter gibberish of the kind I seek out GUIs specifically to avoid having to ever deal with.

Sorry to be the bearer of bad news, but I'm afraid you're going to be stuck with techno-gibberish to get it working, regardless of which program you use. The problem is that all streams from CR are now encrypted. In order to unencrypt them (which is likely not legal in your area), you're going to have to jump through a lot of techno hoops to get the "blob" and "private key" files needed to unencrypt the video stream.

Until a couple days ago, there was still an unencrypted stream that Hama was leveraging to make things work, however that disappeared. Hama is also not willing to venture into potentially illegal territory to provide a means of decryption with his program. I have to respect his stance and decision.

The other links posted are for programs similar to Hama's that do allow you to use your blob and private keys to decrypt videos.  But, the bottom line is that if you want to be able to download and accept the legal responsibility, you will need to generate your own files (per the instructions you mentioned) and use a different program to download. I wouldn't waste your time searching for keys that somebody else generated. Nobody is going to be posting those publicly.

**Comment by NickQ29** [comment]
> > > > > > I mean there is no choice if I want to download anime from crunchy
> > > > > 
> > > > > 
> > > > > Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0
> > > > > Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.
> > > > 
> > > > 
> > > > OR
> > > > https://github.com/Crunchy-DL/Crunchy-Downloader
> > > 
> > > 
> > > well as long as we're all actually linking the projects, https://github.com/anidl/multi-downloader-nx
> > 
> > 
> > Can anyone explain, in terms appropriate for someone with minus infinity tech-savvy, what the difference is between all of these and which one is the easiest to make work for, again, someone with infinitely zero capability for tech-savvy?
> > Because the first one has links to "explanation" threads that are complete and utter gibberish of the kind I seek out GUIs specifically to avoid having to ever deal with.
> 
> Sorry to be the bearer of bad news, but I'm afraid you're going to be stuck with techno-gibberish to get it working, regardless of which program you use. The problem is that all streams from CR are now encrypted. In order to unencrypt them (which is likely not legal in your area), you're going to have to jump through a lot of techno hoops to get the "blob" and "private key" files needed to unencrypt the video stream.
> 
> Until a couple days ago, there was still an unencrypted stream that Hama was leveraging to make things work, however that disappeared. Hama is also not willing to venture into potentially illegal territory to provide a means of decryption with his program. I have to respect his stance and decision.
> 
> The other links posted are for programs similar to Hama's that do allow you to use your blob and private keys to decrypt videos. But, the bottom line is that if you want to be able to download and accept the legal responsibility, you will need to generate your own files (per the instructions you mentioned) and use a different program to download. I wouldn't waste your time searching for keys that somebody else generated. Nobody is going to be posting those publicly.

I actually have a Crunchyroll premium account so (if I understand correctly) I'd be fine using my own keys. The problem is that there _**is**_ no comprehensible explanation for how to generate them.

**Comment by petenun** [comment]
> > > > > > > I mean there is no choice if I want to download anime from crunchy
> > > > > > 
> > > > > > 
> > > > > > Wenn du es riskieren willst: https://github.com/stratuma/Crunchyroll-Downloader-v4.0
> > > > > > Hat auch drm stream support. Ne guide um die keys zu bekommen ist in der readme.
> > > > > 
> > > > > 
> > > > > OR
> > > > > https://github.com/Crunchy-DL/Crunchy-Downloader
> > > > 
> > > > 
> > > > well as long as we're all actually linking the projects, https://github.com/anidl/multi-downloader-nx
> > > 
> > > 
> > > Can anyone explain, in terms appropriate for someone with minus infinity tech-savvy, what the difference is between all of these and which one is the easiest to make work for, again, someone with infinitely zero capability for tech-savvy?
> > > Because the first one has links to "explanation" threads that are complete and utter gibberish of the kind I seek out GUIs specifically to avoid having to ever deal with.
> > 
> > 
> > Sorry to be the bearer of bad news, but I'm afraid you're going to be stuck with techno-gibberish to get it working, regardless of which program you use. The problem is that all streams from CR are now encrypted. In order to unencrypt them (which is likely not legal in your area), you're going to have to jump through a lot of techno hoops to get the "blob" and "private key" files needed to unencrypt the video stream.
> > Until a couple days ago, there was still an unencrypted stream that Hama was leveraging to make things work, however that disappeared. Hama is also not willing to venture into potentially illegal territory to provide a means of decryption with his program. I have to respect his stance and decision.
> > The other links posted are for programs similar to Hama's that do allow you to use your blob and private keys to decrypt videos. But, the bottom line is that if you want to be able to download and accept the legal responsibility, you will need to generate your own files (per the instructions you mentioned) and use a different program to download. I wouldn't waste your time searching for keys that somebody else generated. Nobody is going to be posting those publicly.
> 
> I actually have a Crunchyroll premium account so (if I understand correctly) I'd be fine using my own keys. The problem is that there _**is**_ no comprehensible explanation for how to generate them.

Unfortunately, you understand correctly. If you search, there are other instructions out there on how to generate the files, but none of them are going to be easy. Out of respect to Hama, I will not be posting any links to those sites though. I feel like we're already putting our toes across the line in having this conversation.

**Comment by hama3254** [comment]
> I actually have a Crunchyroll premium account so (if I understand correctly) I'd be fine using my own keys. The problem is that there _**is**_ no comprehensible explanation for how to generate them.

The key talked about is not from Crunchyroll, it's for a encryption method called [widevine](https://developers.google.com/widevine/drm/overview) and is designt to stop what we have been doing.
It's somewhat good at it since i **never** got around it (other than recording) but it can be techical broken via android keys (L3 access) but the media made sure it's illegal (they probably knew there is no 100% safe system) to do so.
I quoted the German law in #918 before the (now gone) switch stream was found.
I don't take any risks with legal stuff because they can destroy your life (like with [gary bowser](https://www.theguardian.com/games/2024/feb/01/the-man-who-owes-nintendo-14m-gary-bowser-and-gamings-most-infamous-piracy-casel)).

The simpelst and legal suggestion (since here the DRM does nothing) is to record the video. This is about 95% as good as a download.


> I would suggest to use OBS and another browser as the default you are using to record the stream rather than paying for a downloading tool.
> 
> More about that here: [#911 (comment)](https://github.com/hama3254/Crunchyroll-Downloader-v3.0/issues/911#issuecomment-2002954344)

**Comment by NickQ29** [comment]
The above is... _sort of_ working (and is better than nothing) but I'm not really sure if I'm doing it right. OBS is barely more newbie-friendly than the technobabble about breaking DRM, which is why I quickly abandoned it as a game recording tool in favor of just using the primitive but absurdly simple Windows Game Bar.

That said this may be off-topic at this point, although if further development has been halted I dunno how much of a problem that is.

**Comment by bybpow** [comment]
I wanted to say that in most countries in the EU breaking DRM is not ilegal for personal purposes, it is in fact legal. But it is against the TOS of the platforms so they can ban you although they never do it in due to the complication of noticing what users are doing it and what users are not . This is not the case for Germany and Austria that have really strong copyright laws, even torrenting is tricky there. I'd love Hama to implement the DRM bypass but I'd not recommend it. There are already tools that do this although is more tech savvy. There are other options more easy but are paid.

So, are we at the end of this repo?

**Comment by stratuma** [comment]
> I wanted to say that in most countries in the EU breaking DRM is not ilegal for personal purposes, it is in fact legal. But it is against the TOS of the platforms so they can ban you although they never do it due to the complication of noticing what users are doing it and what users are not . This is not the case for Germany and Austria that have really strong copyright laws, even torrenting is tricky there. I'd love Hama to implement the DRM bypass but I'd not recommend it. There are already tools that do this although is more tech savvy. There are other options more easy but are paid.
> 
> So, are we at the end of this repo?

If anybody outside of germany or any drm problematic country starts to mantain it it could still stay active

**Comment by TaMaGo-TTV** [comment]
> > I wanted to say that in most countries in the EU breaking DRM is not ilegal for personal purposes, it is in fact legal. But it is against the TOS of the platforms so they can ban you although they never do it due to the complication of noticing what users are doing it and what users are not . This is not the case for Germany and Austria that have really strong copyright laws, even torrenting is tricky there. I'd love Hama to implement the DRM bypass but I'd not recommend it. There are already tools that do this although is more tech savvy. There are other options more easy but are paid.
> > So, are we at the end of this repo?
> 
> If anybody outside of germany or any drm problematic country starts to mantain it it could still stay active

Does that mean that the users who decrypt video with the l3 keys are safe? But the person who offers the downloader will be prosecuted?

**Comment by bybpow** [comment]
> > > I wanted to say that in most countries in the EU breaking DRM is not ilegal for personal purposes, it is in fact legal. But it is against the TOS of the platforms so they can ban you although they never do it due to the complication of noticing what users are doing it and what users are not . This is not the case for Germany and Austria that have really strong copyright laws, even torrenting is tricky there. I'd love Hama to implement the DRM bypass but I'd not recommend it. There are already tools that do this although is more tech savvy. There are other options more easy but are paid.
> > > So, are we at the end of this repo?
> > 
> > 
> > If anybody outside of germany or any drm problematic country starts to mantain it it could still stay active
> 
> Does that mean that the users who decrypt video with the l3 keys are safe? But the person who offers the downloader will be prosecuted?

Users are mostly safe unless you are from Germany or Austria. You may also be safe there but you could get in trouble.  Developers on the other side are prosecuted more often, specially there. You can decrypt your own L3 keys but you will need to "through away your phone" as you will need to unroot and extract the CDM. If they catch you key and ban you the phone is gone forever as Widevine is linked to the device ID.

**Comment by TaMaGo-TTV** [comment]
> > > > I wanted to say that in most countries in the EU breaking DRM is not ilegal for personal purposes, it is in fact legal. But it is against the TOS of the platforms so they can ban you although they never do it due to the complication of noticing what users are doing it and what users are not . This is not the case for Germany and Austria that have really strong copyright laws, even torrenting is tricky there. I'd love Hama to implement the DRM bypass but I'd not recommend it. There are already tools that do this although is more tech savvy. There are other options more easy but are paid.
> > > > So, are we at the end of this repo?
> > > 
> > > 
> > > If anybody outside of germany or any drm problematic country starts to mantain it it could still stay active
> > 
> > 
> > Does that mean that the users who decrypt video with the l3 keys are safe? But the person who offers the downloader will be prosecuted?
> 
> Users are mostly safe unless you are from Germany or Austria. You may also be safe there but you could get in trouble. Developers on the other side are prosecuted more often, specially there. You can decrypt your own L3 keys but you will need to "through away your phone" as you will need to unroot and extract the CDM. If they catch you key and ban you the phone is gone forever as Widevine is linked to the device ID.

I mean the Key is easy to get with android Studio... So thats not the Problem. But how are they tracking down the users? 

I live in Germany my only options are OBS, L3 Key or Torrent.

**Comment by bybpow** [comment]
> > > > > I wanted to say that in most countries in the EU breaking DRM is not ilegal for personal purposes, it is in fact legal. But it is against the TOS of the platforms so they can ban you although they never do it due to the complication of noticing what users are doing it and what users are not . This is not the case for Germany and Austria that have really strong copyright laws, even torrenting is tricky there. I'd love Hama to implement the DRM bypass but I'd not recommend it. There are already tools that do this although is more tech savvy. There are other options more easy but are paid.
> > > > > So, are we at the end of this repo?
> > > > 
> > > > 
> > > > If anybody outside of germany or any drm problematic country starts to mantain it it could still stay active
> > > 
> > > 
> > > Does that mean that the users who decrypt video with the l3 keys are safe? But the person who offers the downloader will be prosecuted?
> > 
> > 
> > Users are mostly safe unless you are from Germany or Austria. You may also be safe there but you could get in trouble. Developers on the other side are prosecuted more often, specially there. You can decrypt your own L3 keys but you will need to "through away your phone" as you will need to unroot and extract the CDM. If they catch you key and ban you the phone is gone forever as Widevine is linked to the device ID.
> 
> I mean the Key is easy to get with android Studio... So thats not the Problem. But how are they tracking down the users?
> 
> I live in Germany my only options are OBS, L3 Key or Torrent.

They ask your ISP for your IP mostly. They also request VPNs for your IPs in case you use them. It is the most used case. I would not recommend OBS due to the size and quality or playback issues. DRM usually block screen recording too but for an archivist datahoarder as myself I would always try DL before recording.

**Comment by TaMaGo-TTV** [comment]
> > > > > If anybody outside of germany or any drm problematic country starts to mantain it it could still stay active
> > > > 
> > > > 
> > > > Does that mean that the users who decrypt video with the l3 keys are safe? But the person who offers the downloader will be prosecuted?
> > > 
> > > 
> > > Users are mostly safe unless you are from Germany or Austria. You may also be safe there but you could get in trouble. Developers on the other side are prosecuted more often, specially there. You can decrypt your own L3 keys but you will need to "through away your phone" as you will need to unroot and extract the CDM. If they catch you key and ban you the phone is gone forever as Widevine is linked to the device ID.
> > 
> > 
> > I mean the Key is easy to get with android Studio... So thats not the Problem. But how are they tracking down the users?
> > I live in Germany my only options are OBS, L3 Key or Torrent.
> 
> They ask your ISP for your IP mostly. They also request VPNs for your IPs in case you use them. It is the most used case. I would not recommend OBS due to the size and quality or playback issues. DRM usually block screen recording too but for an archivist datahoarder as myself I would always try DL before recording.

So I guess DL with VPN (Provider with nolog policy) should be the best and "safest" Option 🤔

I rly hate crunchy for that shit man

**Comment by stratuma** [comment]
Video decryption is way more safe than torrenting

**Comment by Serpentstar** [comment]
NordVPN states they do not keep logs and I have used them for years.  The thing I really like about them is they have a Linux flavor, and with a bit of effort, can even be ran on a Raspberry PI.

Can someone here provide a comprehensive list of which DL methods would work best?  Paid applications or not.  There have been a bunch of ones listed, but they are all over the place.  I just want something easy to use that can get one episode at a time.

Thanks,.

**Comment by stratuma** [comment]
NordVPN, u joking xDDDDD

**Comment by TaMaGo-TTV** [comment]
> NordVPN, u joking xDDDDD

NordVPN is a crazy call 😂😂😂😂😂😂

**Comment by petenun** [comment]
I don't know if there's a better place to hold this conversation. I don't know GitHub's rules but I wouldn't want to get Hama into trouble talking casually about circumventing laws. Yeah, I know. I'm a stick in the mud.

**Comment by TaMaGo-TTV** [comment]
> I don't know if there's a better place to hold this conversation. I don't know GitHub's rules but I wouldn't want to get Hama into trouble talking casually about circumventing laws. Yeah, I know. I'm a stick in the mud.

Yeah lets stop this here and close this thread until there is a Fix (Hopefully) 🤝🏻

**Comment by stratuma** [comment]
> @stratuma @TaMaGo-TTV Enlighten me please. DM me if you don't mind.

You are safer without VPN instead of using NordVPN tbh, they track everything and work with authorities. (Thats why they are so cheap)

**Comment by NickQ29** [comment]
Actually, before this is closed - is it still possible to download just the subtitles, the way it was the last time this happened?

**Comment by hama3254** [comment]
> Actually, before this is closed - is it still possible to download just the subtitles, the way it was the last time this happened?

the latest release should work fine for subtitles. The add window has a drop down to just download them.

**Comment by TaMaGo-TTV** [comment]
> > Actually, before this is closed - is it still possible to download just the subtitles, the way it was the last time this happened?
> 
> the latest release should work fine for subtitles. The add window has a drop down to just download them.

Not working for me

**Comment by hama3254** [comment]
I see no issue on my side. You can check if you have subtitles on CR itself new episodes and dubs may simply have none.
The downloader does not issue any error for missing soft subs.
![grafik](https://github.com/hama3254/Crunchyroll-Downloader-v3.0/assets/52826614/47bb2569-e713-4ef9-b965-c42974a3c6a7)

**Comment by Akamaru** [comment]
If someone want to download just the subtitles, I created a .bat file to do this.

```batch
@echo off
:loop
set /p url="Bitte gib die Crunchyroll-URL ein (oder 'exit' zum Beenden): "
if "%url%"=="exit" goto :eof
set flags=--allow-u --write-sub --sub-lang de-DE,en-US --sub-format "ass" -u "EMAIL" -p "PASSWORD" --skip-download
cd /d "FULL PATH TO DESTINATION"
yt-dlp %flags% %url%
goto :loop
```
You need yt-dlp installed and added to your PATH in windows.

And to download anime from ADN I have this .bat:

```batch
@echo off
:loop
set /p url="Bitte gib die ADN-URL ein (oder 'exit' zum Beenden): "
if "%url%"=="exit" goto :eof
set flags=--sub-lang "de" --embed-subs --remux-video mkv -u "EMAIL" -p "PASSWORD" --embed-metadata
cd /d "FULL PATH TO DESTINATION"
yt-dlp %flags% %url%
goto :loop
```

Important: If your password contains a % you need to use %%.

**Comment by acteiaaaaaaaaaaaaaaaaaaa** [comment]
It's really sad to see the program I used for years being discontinued. No other program surpasses this beautiful work, so simple and unique. Thank you for dedicating all this time; it will leave a huge hole.😭

**Comment by Denji945** [comment]
@hama3254 So this is the end of the downloader?

**Comment by sahtanoj** [comment]
I don't believe this is the end of CRD, as I believe Hama will find some way around it, even if it's through the insertion of a DRM unlock. However, if this is the end, @hama3254 , thank you very much for all this incredible work. You will be in all of our hearts.❤️

**Comment by petenun** [comment]
I agree with your sentiment about Hama and all his work. It's been a fun ride. Though I obviously can't speak for him, in the past he's made it pretty clear that he's not willing to cross the line of dealing with DRM, and I certainly wouldn't want to push him in a direction in which he's not willing to go. That said, it's his decision to make. Whatever happens, I'm still thankful for all the hard work they've put into this project.

**Comment by hama3254** [comment]
> as I believe Hama will find some way around it, even if it's through the insertion of a DRM unlock.
 

As already stated I don't breake the DRM more here: https://github.com/hama3254/Crunchyroll-Downloader-v3.0/issues/991#issuecomment-2181970353
